### PR TITLE
Unlink translations from each other

### PR DESCRIPTION
**Pull request by pcraig3**
# Summary | Résumé

This PR introduces a new API route and some new logic in our PageSelect component to call the new route if we select "None" as the page translation. It also adds some tests, which brings up our total coverage up to 67%.

Routes we have:

1. Set a translation: `POST` -> `http://localhost/wp-json/cds/wpml/posts/47/translation` (param: `translationId`)
2. Get a translation: `GET` -> `http://localhost/wp-json/cds/wpml/posts/47/translation`
3. (New ✨) Unset a translation: `DELETE` -> `http://localhost/wp-json/cds/wpml/posts/47/translation`

It's not technically a `DELETE`, since no resource is actually getting deleted, but Dave and I talked about it and it was the most intuitive route we came up with.

## gif

Screen Recording 2022-08-03 at 11 08 39

### CODE REVIEW

File: wordpress/wp-content/plugins/cds-wpml-mods/resources/js/util/fetch.js [hunk 1]
```diff
@@ -21,14 +21,14 @@ export const getData = async (endpoint) => {
...
-export const sendData = async (endpoint, data) => {
+export const sendData = async ({ endpoint, data, method }) => {
     const requestHeaders = new Headers({
         'Content-Type': 'application/json;charset=UTF-8',
     });
     requestHeaders.append('X-WP-Nonce', CDS_VARS.rest_nonce);
 
     const response = await fetch(`${CDS_VARS.rest_url}${endpoint}`, {
-        method: 'POST',
+        method: method,
```
Comment (timarney) [inline wordpress/wp-content/plugins/cds-wpml-mods/resources/js/util/fetch.js:31]: We should validate the method here so an arbitrary value isn't passed.

File: wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php [hunk 2]
```diff
@@ -191,4 +201,38 @@ public function getTranslation(WP_REST_Request $request): WP_REST_Response|WP_Er
...
+     * Get the existing translation for a given page.
+     * Returns a WP_Error if no post is found for the id given.
+     *
+     * @param  WP_REST_Request  $request
+     *
+     * @return array | WP_Error
+     */
+    public function unsetTranslation(WP_REST_Request $request)// : WP_REST_Response|WP_Error
+    {
+        $post = get_post($request['id']);
```
Comment (timarney) [inline wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php:215]: This should be changed to `intval($request['id'])`

Comment (timarney) [inline wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php:215]:
```suggestion
        $post = get_post(intval($request['id']));
```

File: wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php [hunk 3]
```diff
@@ -191,4 +201,38 @@ public function getTranslation(WP_REST_Request $request): WP_REST_Response|WP_Er
...
+
+        if (is_null($post)) {
+            return new WP_Error('post_not_found', __('The post you are looking for does not exist', 'cds-wp-mods'), array( 'status' => 404 ));
+        }
+
+        $postType = 'post_' . $post->post_type;
+        $language_code = $this->post->getLanguageCodeOfPostObject($post);
+
+        // Unset existing translation if exists
+        if ($translatedPostID = $this->post->getTranslatedPostID($post)) {
```
Comment (timarney) [inline wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php:225]: Might want to wrap this is a try / catch --- to ensure the `200` is actually valid

File: wordpress/wp-content/plugins/cds-wpml-mods/resources/js/util/fetch.js (hunk 1 above)
Comment (pcraig3) [reply wordpress/wp-content/plugins/cds-wpml-mods/resources/js/util/fetch.js]: Makes sense, updated.

File: wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php (hunk 2 above)
Comment (pcraig3) [reply wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php]: Added it everywhere we use "get_post"

File: wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php (hunk 3 above)
Comment (pcraig3) [reply wordpress/wp-content/plugins/cds-wpml-mods/src/Api/Endpoints.php]: That works, added it to all of our calls.

**Review by timarney — APPROVED** [review]

**Pull request merged**